docs/migration.rst

```diff
@@ -0,0 +1,368 @@
+Migration Guide
+===============
+
+This guide explains how to migrate from using ``pytz`` to a
+:pep:`495`-compatible library, either the standard library module
+:mod:`zoneinfo` (or its backport, :pypi:`backports.zoneinfo`), or
+:pypi:`python-dateutil`.
+
+Which replacement to choose?
+----------------------------
+
+If you only need to support Python 3, you should use :mod:`zoneinfo`. If you
+need to support Python 2 and Python 3, you should either use
+:mod:`dateutil.tz` or a combination of ``zoneinfo`` and ``dateutil``. The
+shim classes use ``dateutil`` in Python 2 and ``zoneinfo`` in Python 3.
+
+You can also directly use the shim classes if desired; the shim classes are,
+themselves, :pep:`495`-compatible time zone implementations. As long as you do
+not use any part of the ``pytz``-specific interface, they will not raise any
+warnings. There is, however, some performance overhead associated with this as
+compared to using the underlying libraries directly.
+
+If your time zones are coming from a source that returns
+``pytz_deprecation_shim`` shim time zones, you can upgrade them to the PEP
+495-compatible zone that they are a wrapper around using
+:func:`pytz_deprecation_shim.helpers.upgrade_tzinfo`.
+
+For more information about how to get time zones from the replacement you've
+chosen, see the section on `constructing a tzinfo object
+<constructing-tzinfo>`_.
+
+Creating an aware datetime ("localizing" datetimes)
+---------------------------------------------------
+
+With :pep:`495`-compatible zones, there is no longer any need to use
+``localize``, you can directly attach a time zone to your ``datetime``:
+
+.. code-block:: python
+
+    import zoneinfo
+    from datetime import datetime
+
+    NYC = zoneinfo.ZoneInfo("America/New_York")
+    datetime(2020, 1, 1, tzinfo=NYC)
+
+If you have a naïve ``datetime`` (or one you'd like to "reinterpret" as being
+in another zone), use ``replace``:
+
+.. code-block:: python
+
+    datetime(2020, 1, 1).replace(tzinfo=NYC)
+
+Ambiguous and imaginary times
+#############################
+
+Whenever a time zone's UTC offset changes (such as during a Daylight Saving
+Time transition), this creates either a gap ("imaginary" times) of times that
```

Comment:
So DST is called "Summer Time" outside of the USA/Canada, not sure if you like to throw an "aka" in there, for l10n and/or SEO.